x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_store_data_client.ts

```diff
@@ -79,18 +94,161 @@ export class EntityStoreDataClient {
+    await this.createEntityIndex(entityType);
+    debugLog(`Created entity index`);
+
+    // we must create and execute the enrich policy before the pipeline is created
+    // this is because the pipeline will fail if the enrich index does not exist
+    await this.createFieldRetentionEnrichPolicy(entityType);
+    debugLog(`Created field retention enrich policy`);
+    await this.executeFieldRetentionEnrichPolicy(entityType);
+    debugLog(`Executed field retention enrich policy`);
+    await this.createPlatformPipeline(entityType);
+    debugLog(`Created @platform pipeline`);
+
+    // finally start the entity definition now that everything is in place
+    await this.start(entityType, { force: true });
+    debugLog(`Started entity definition`);
+
+    // the task will execute the enrich policy on a schedule
+    await startEntityStoreFieldRetentionEnrichTask({
+      namespace: this.options.namespace,
+      logger,
+      taskManager,
+    });
+    debugLog(`Started entity store field retention enrich task`);
 
+    // and finally update the engine status to started once everything is in place
+    const updated = await this.engineClient.update(definition.id, ENGINE_STATUS.STARTED);
+    debugLog(`Updated engine status to 'started', initialisation complete`);
+    logger.info(`Entity store initialized`);
     return { ...descriptor, ...updated };
   }
 
-  public async start(entityType: EntityType) {
-    const definition = getEntityDefinition(entityType, this.options.namespace);
+  public async executeFieldRetentionEnrichPolicy(
+    entityType: EntityType
+  ): Promise<{ executed: boolean }> {
+    try {
+      await executeFieldRetentionEnrichPolicy({
+        namespace: this.options.namespace,
+        esClient: this.options.esClient,
+        entityType,
+      });
+      return { executed: true };
+    } catch (e) {
+      if (e.statusCode === 404) {
+        return { executed: false };
+      }
+      this.options.logger.error(
+        `Error executing field retention enrich policy for ${entityType}: ${e.message}`
+      );
+      throw e;
+    }
+  }
+
+  private async createFieldRetentionEnrichPolicy(entityType: EntityType) {
+    return createFieldRetentionEnrichPolicy({
+      namespace: this.options.namespace,
+      esClient: this.options.esClient,
+      entityType,
+    });
+  }
+
+  private async deleteFieldRetentionEnrichPolicy(entityType: EntityType) {
+    return deleteFieldRetentionEnrichPolicy({
+      namespace: this.options.namespace,
+      esClient: this.options.esClient,
+      entityType,
+    });
+  }
+
+  private async createPlatformPipeline(entityType: EntityType) {
+    const definition = getDefinitionForEntityType(entityType, this.options.namespace);
+
+    const allEntityFields: string[] = (definition?.metadata || []).map((m) => {
+      if (typeof m === 'string') {
+        return m;
+      }
+
+      return m.destination;
+    });
+
+    const pipeline = {
+      id: `${definition.id}-latest@platform`,
```

Comment:
Do we share the same ingest pipeline for all spaces? Or are all resources unique by space?